ibm_cloud_sdk_core/utils.py

```diff
@@ -125,15 +136,15 @@ def read_from_credential_file(service_name, separator='='):
     return config
 
 def _parse_key_and_update_config(config, service_name, key, value):
-    if service_name in key:
+    if key.startswith(service_name):
         index = key.find('_')
```

Comment:
I don't think this works if the service_name contains an "_" does it???

You can see the logic that's used by the java core in `CredentialUtils.parseCredentials()`, to parse a line from either a credential file or an env var setting, which looks like this:
```
serviceName = serviceName.toUpperCase().replaceAll("-", "_") + "_";
...
if (key.startsWith(serviceName)) {
  String credentialName = key.substring(serviceName.length());
  ...
}
```

Reply:
This will work as before calling the `_parse_key_and_update_config `, the calling function perform's the replace

Reply:
The service_name passed into `_parse_key_and_update_config` could have an "_" in it, so this code fragment won't work:
```
    if key.startswith(service_name):
        index = key.find('_')
        if index != -1:
            config[key[index + 1:]] = value
```
If this code is processing a key of "personality_insights_url", you'd end up with a config map with this entry:
```
   "insights_url" --> <url value>
```
Right?

Please add a test that includes a multi-word service name and you will find out exactly what happens.

Reply:
Ahaaa I see that the truncation happening here! Good catch! 👍